lua/toggleterm/ui.lua

```diff
@@ -77,31 +77,29 @@ end
 ---apply highlights to a terminal
 ---@param term Terminal
 function M.hl_term(term)
-  local highlights = {}
-  local config = require("toggleterm.config")
-  if term and term:is_float() or M.is_float() then
-    local opts = term and term.float_opts or config.get("float_opts") or {}
-    highlights = {
-      fmt("NormalFloat:%s", opts.highlights.background),
-      fmt("FloatBorder:%s", opts.highlights.border),
-    }
-  elseif not vim.tbl_isempty(term.highlights) then
-    for key, value in pairs(term.highlights) do
-      highlights[#highlights + 1] = key .. ":" .. value
-    end
-  elseif config.get("shade_terminals") then
-    highlights = {
-      "Normal:DarkenedPanel",
-      "EndOfBuffer:DarkenedPanel",
-      "VertSplit:DarkenedPanel",
-      "StatusLine:DarkenedStatusline",
-      "StatusLineNC:DarkenedStatuslineNC",
-      "SignColumn:DarkenedPanel",
-    }
-  end
-  if highlights then
-    vim.cmd("setlocal winhighlight=" .. table.concat(highlights, ","))
+  if type(term.highlights) ~= "table" or vim.tbl_isempty(term.highlights) then
+    return
   end
+
+  local highlights = vim.tbl_map(function(hl_group_name)
+    local hi_def = constants.highlight_group_name_prefix .. term.id .. hl_group_name
+    local hi_target = hl_group_name .. ":" .. hi_def
+
+    local hi_attrs = term.highlights[hl_group_name]
+
+    if hi_attrs.link then
+      vim.cmd(fmt("highlight default link %s %s", hi_def, hi_attrs.link))
```

Comment:
There is actually a private module you can currently use for highlighting in nvim called `vim.highlight` it has the methods `vim.highlight.create` and `vim.highlight.link` I think those have nicer semantics than  constructing the string hl commands yourself. It's a temporary solution anyway, since once 0.7 is out I'll migrate this to `nvim_set_hl`. But for now I think this will be better as
```lua
vim.highlight.create('Name', {guibg = 'Color', gui = 'GUI'})
```